pkg/cvo/sync_worker.go

```diff
@@ -493,6 +504,30 @@ func (w *SyncWorker) syncOnce(ctx context.Context, work *SyncWork, maxWorkers in
 		payloadUpdate.VerifiedImage = info.Verified
 		payloadUpdate.LoadedAt = time.Now()
 
+		// need to make sure the payload is only set when the preconditions have been successfull
+		if !info.Local && len(w.preconditions) > 0 {
+			reporter.Report(SyncWorkerStatus{
```

Comment:
I'm confused, shouldn't this be an else block?

Reply:
was following https://github.com/openshift/cluster-version-operator/pull/243/files/64249a4e283ddf99631885faedf6297401fb7e88#diff-b2099007e7494e37fc4f7a4320f3622bR471-R489

where we send update that we are running the preconditions and and a following failure update if failed to do so..